addons/binding/org.openhab.binding.phc/src/main/java/org/openhab/binding/phc/handler/PHCBridgeHandler.java

```diff
@@ -0,0 +1,693 @@
+                                                                    // PHC didn't recognize it.
+                    } else {
+                        sendEmAcknowledge(moduleAddress, toggleIn);
+                        handleIncomingCommand(moduleAddress, channel[0], cmd);
+
+                    }
+                }
+            }
+        }
+
+        if (logger.isDebugEnabled()) {
+            StringBuilder logMessage = new StringBuilder();
+            for (int i = 0; i < inByte.length; i++) {
+                logMessage.append(PHCHelper.byteToBinaryString(inByte[i]));
+            }
+            logMessage.append(", " + inByte.length);
+            logger.debug("received: {}", logMessage);
+        }
+    }
+
+    /**
+     * Send the incoming command to the appropriate handler and channel.
+     *
+     * @param moduleAddress
+     * @param channel
+     * @param cmd
+     * @param rcvCrc
+     */
+    private void handleIncomingCommand(byte moduleAddress, int channel, byte[] cmd) {
+        ThingUID uid = PHCHelper.getThingUIDreverse(PHCBindingConstants.THING_TYPE_EM, moduleAddress);
+        Thing thing = getThingByUID(uid);
+        String channelId = "em#" + StringUtils.leftPad(Integer.toString(channel), 2, '0');
+
+        if (thing != null) {
+            OnOffType state = OnOffType.OFF;
+            if ((cmd[0] & 0x0F) == 2) {
+                state = OnOffType.ON;
+            }
+
+            logger.debug("{}, {}", thing.getUID(), state);
+            ((PHCHandler) thing.getHandler()).handleIncoming(channelId, state);
+
+        } else {
+            logger.info("No Thing with UID {} available.", uid.getAsString());
+        }
+    }
+
+    /**
+     * Send message to to serial port x times.
+     *
+     * @param msg
+     * @param sendCnt
+     */
+    private void writeMsg(byte[] msg, int sendCnt) {
+        while (0 < sendCnt) {
+            serialWrite(msg);
+            sendCnt--;
+        }
+    }
+
+    private void serialWrite(byte[] msg) {
+        try {
+            // write to serial port
+            serialOut.write(msg);
+            serialOut.flush();
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Error writing '" + msg + "' to serial port : " + e.getMessage());
+        }
+
+        if (logger.isDebugEnabled()) {
+            StringBuilder log = new StringBuilder();
+            for (byte b : msg) {
+                log.append(PHCHelper.byteToBinaryString(b));
+                log.append(' ');
+            }
+            logger.debug("send: {}", log);
+        }
+    }
+
+    /**
+     * Adds the given address to the module list.
+     *
+     * @param module
+     */
+    public void addModule(byte module) {
+        modules.add(module);
+    }
+
+    /**
+     * Object to save a whole message.
+     *
+     * @author Jonas Hohaus
+     */
+    class QueueObject {
```

Comment:
Since the class is already big, maybe just put this outside the class but in the same package.